src/util/useSortTableData.tsx

```diff
@@ -0,0 +1,45 @@
+import { SortDirection } from "@canonical/react-components";
+import { useState } from "react";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+
+type Props = {
+  rows: MainTableRow[];
```

Comment:
```suggestion
  rows: unknown[];
```

Reply:
this hook does rely on the MainTableRow data format since we need the sortData field from rows objects